jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/RegexPathSpec.java

```diff
@@ -357,78 +371,92 @@ private class RegexMatchedPath implements MatchedPath
     {
         private final RegexPathSpec pathSpec;
         private final String path;
-        private final Matcher matcher;
-
-        private final String pathMatch;
-        private final String pathInfo;
+        private String pathMatch;
+        private String pathInfo;
 
         public RegexMatchedPath(RegexPathSpec regexPathSpec, String path, Matcher matcher)
         {
             this.pathSpec = regexPathSpec;
             this.path = path;
-            this.matcher = matcher;
 
+            calcPathMatchInfo(matcher);
+        }
+
+        private void calcPathMatchInfo(Matcher matcher)
+        {
             int groupCount = matcher.groupCount();
 
-            int idxNameStart = startOf(matcher, "name");
-            int idxNameEnd = endOf(matcher, "name");
-            int idxInfoStart = startOf(matcher, "info");
-            int idxInfoEnd = endOf(matcher, "info");
 
             if (groupCount == 0)
             {
                 pathMatch = path;
                 pathInfo = null;
+                return;
             }
-            else if (groupCount == 1)
+
+            if (groupCount == 1)
             {
-                if (idxNameStart >= 0)
+                // we know we are splitting
+                int idxNameEnd = endOf(matcher, "name");
```

Comment:
Ultimately this lookup makes me itch!   If they are not using the named group, then an exception is created and ignored for every lookup attempt.    

For now, can we add a TODO that when we scan the pattern, we should look for the presence of "<name>" and only do this lookup if we have seen it in the pattern.